ra_aid/agent_utils.py

```diff
@@ -208,22 +225,22 @@ def run_web_research_agent(
     """
     thread_id = thread_id or str(uuid.uuid4())
     logger.debug("Starting web research agent with thread_id=%s", thread_id)
-    logger.debug("Web research configuration: expert=%s, hil=%s, web=%s",
-                expert_enabled, hil, web_research_enabled)
 
     # Initialize memory if not provided
     if memory is None:
         memory = MemorySaver()
 
-    # Set up thread ID
-    if thread_id is None:
-        thread_id = str(uuid.uuid4())
-
-    # Configure tools using restricted web research toolset
-    tools = get_web_research_tools(expert_enabled=expert_enabled)
+    # Configure tools
+    tools = get_web_research_tools(
+        expert_enabled=expert_enabled,
+        human_interaction=hil,
+        web_research_enabled=web_research_enabled
+    )
 
-    # Create agent
-    agent = create_react_agent(model, tools, checkpointer=memory)
+    # Create agent if model is provided
+    agent = None
+    if model is not None:
```

Comment:
Related to above comment on __main__, I think we won't actually have any case where model is None.